spec/rails_config_validator/validator_spec.rb

```diff
@@ -0,0 +1,306 @@
+require 'spec_helper'
+
+describe RailsConfigValidator::Validator do
+  let(:dir) { Dir.mktmpdir }
+  let(:schema_path) { File.join(dir, "#{config_name}.schema.yml") }
+  let(:config_path) { File.join(dir, "#{config_name}.yml") }
+  let(:config_name) { 'config' }
+  let(:env) { 'test' }
+  let(:schema) { margin_schema.gsub(/^\s*\|/, '') }
+  let(:config) { margin_config.gsub(/^\s*\|/, '') }
+
+  shared_examples_for 'valid configuration' do
+    describe '#valid?' do
+      it 'is invalid' do
+        expect(validator.valid?).to eq(true)
+      end
+    end
+
+    describe '#valid!' do
+      it 'raises error' do
+        expect { validator.valid! }.to_not raise_error
+      end
+    end
+  end
+
+  shared_examples_for 'invalid configuration' do
+    describe '#valid?' do
+      it 'is invalid' do
+        expect(validator.valid?).to eq(false)
+      end
+    end
+
+    describe '#valid!' do
+      it 'raises error' do
+        expect { validator.valid! }.to raise_error(RailsConfigValidator::InvalidConfigurationError)
+      end
+    end
+  end
+
+  shared_examples_for 'invalid schema' do
+    describe '#valid?' do
+      it 'is invalid' do
+        expect(validator.valid?).to eq(false)
+      end
+    end
+
+    describe '#valid!' do
+      it 'raises error' do
+        expect { validator.valid! }.to raise_error(RailsConfigValidator::InvalidSchemaError)
+      end
+    end
+  end
+
+  describe '#new' do
+    context 'when only config path is passed' do
+      subject { described_class.new(config_path, env) }
+
+      it 'creates new object' do
+        expect(subject).to be_truthy
+      end
+    end
+
+    context 'when schema path is passed explicitly' do
+      let(:schema_path2) { File.join(dir, "#{config_name}2.schema.yml") }
+      subject { described_class.new(config_path, env, schema_path: schema_path2) }
+
+      it 'creates new object' do
+        expect(subject).to be_truthy
```

Comment:
Michal's testing if ruby runtime works properly ;P
When testing constructor, I'd only test if it fails when it's designed to fail in certain cases. Constructor success paths should already be covered in other test cases for "normal" methods.